awx/main/expect/isolated_manager.py

```diff
@@ -1,162 +1,112 @@
-        that manages the isolated instance.
-        Use of normal job callback and other such configurations are avoided.
-        '''
+    def build_runner_params(self, hosts, verbosity=1):
         env = dict(os.environ.items())
         env['ANSIBLE_RETRY_FILES_ENABLED'] = 'False'
         env['ANSIBLE_HOST_KEY_CHECKING'] = 'False'
         env['ANSIBLE_LIBRARY'] = os.path.join(os.path.dirname(awx.__file__), 'plugins', 'isolated')
-        return env
+        set_pythonpath(os.path.join(settings.ANSIBLE_VENV_PATH, 'lib'), env)
 
-    @staticmethod
-    def _build_args(playbook, hosts, extra_vars=None):
-        '''
-        Returns list of Ansible CLI command arguments for a management task
-
-        :param playbook:   name of the playbook to run
-        :param hosts:      host pattern to operate on, ex. "localhost,"
-        :param extra_vars: optional dictionary of extra_vars to apply
-        '''
-        args = [
-            'ansible-playbook',
-            playbook,
-            '-u', settings.AWX_ISOLATED_USERNAME,
-            '-T', str(settings.AWX_ISOLATED_CONNECTION_TIMEOUT),
-            '-i', hosts
-        ]
-        if extra_vars:
-            args.extend(['-e', json.dumps(extra_vars)])
-        if settings.AWX_ISOLATED_VERBOSITY:
-            args.append('-%s' % ('v' * min(5, settings.AWX_ISOLATED_VERBOSITY)))
-        return args
-
-    @classmethod
-    def awx_playbook_path(cls):
-        return os.path.abspath(os.path.join(
-            os.path.dirname(awx.__file__),
-            'playbooks'
-        ))
+        def finished_callback(runner_obj):
+            if runner_obj.status == 'failed':
+                stdout = runner_obj.stdout.read()
+                playbook_logger.error(stdout)
+                event_data = {'event': 'verbose', 'stdout': stdout, self.event_data_key: self.instance.id}
+                CallbackQueueDispatcher().dispatch(event_data)
+            else:
+                playbook_logger.info(runner_obj.stdout.read())
+
+        inventory = '\n'.join([
+            '{} ansible_ssh_user={}'.format(host, settings.AWX_ISOLATED_USERNAME)
+            for host in hosts
+        ])
+
+        return {
+            'project_dir': os.path.abspath(os.path.join(
+                os.path.dirname(awx.__file__),
+                'playbooks'
+            )),
+            'inventory': inventory,
+            'envvars': env,
+            'finished_callback': finished_callback,
+            'verbosity': verbosity,
+            'cancel_callback': self.cancelled_callback,
+            'settings': {
+                'idle_timeout': self.idle_timeout,
+                'job_timeout': settings.AWX_ISOLATED_LAUNCH_TIMEOUT,
+                'pexpect_timeout': getattr(settings, 'PEXPECT_TIMEOUT', 5),
+            },
+        }
 
     def path_to(self, *args):
         return os.path.join(self.private_data_dir, *args)
 
+    def run_management_playbook(self, playbook, private_data_dir, **kw):
```

Comment:
Can we list all the explicitly list all the kwargs that are expected?

Reply:
eeeeeeeh?  This could be a _really_ long list, and it's subject to change/grow with each version of runner:

https://github.com/ansible/ansible-runner/blob/master/ansible_runner/interface.py#L74